docs: update vscode docs for devkit compatibility (#14815)
unoplatform/uno: doc/articles/get-started-vscode-devkit.md

Reviewer: This file and the other new doc file are missing in toc

Reviewer: Probably both new articles should link to each other.

Author: there's a _aka_ link (used by the extension) to the parent document describing the pros/cons of each. it's probably better to link to that

---
uid: Uno.GetStarted.vscode.DevKit
---

> ## How to switch to C# DevKit Mode

Reviewer: This should be an h1, I think. And next headings should also be reduced by one level

Author: yes, that would be the correct markdown
the structure was copied from existing documents and, since it's processed, I'm not sure of end-result. Being identical to other documents was safer.

> Note: Due to the [unavailability](https://github.com/microsoft/vscode-dotnettools/issues/765) of the **C# DevKit extension** on [OpenVSX](https://open-vsx.org) users of VSCodium, Gitpod... are always on the OmniSharp mode.

Reviewer: ```suggestion
Note: Due to the [unavailability](https://github.com/microsoft/vscode-dotnettools/issues/765) of the **C# DevKit extension** on [OpenVSX](https://open-vsx.org), users of VSCodium, Gitpod... are always on the OmniSharp mode.
```

### Ensure you are using either .NET 8 (or later) or Uno.WinUI 5.0.116 (or later)

> To support C# DevKit the Uno Platform extension needs to run on .NET 8 (or later) or use Uno vFIXME (or later).

Reviewer: ```suggestion
To support C# DevKit the Uno Platform extension needs to run on .NET 8 (or later) or use Uno 5.0.116 (or later).
```

#### .NET 8

From a terminal (external or inside VS Code) type:

```bash
dotnet --version
```

If the version listed is older than `8.0.100` then, inside VS Code, press `F1` and select `Run Uno-Check to setup this environment for Uno Platform` to update your system. Quit and restart VS Code afterward so the new version of `dotnet` can be used by the extensions.

If you have version `8.0.100` (or later) then there's no need to update your version of .NET or Uno.

#### Uno.WinUI 5.0.116 or later

> The latest versions of Uno ships with an additional msbuild task so running .NET 8 is not required.

Reviewer: ```suggestion
The latest versions of Uno ships with an additional MSBuild task, so running .NET 8 is not required.
```

> To see which version of Uno you're currently using open the `Directory.Packages.props` file at the root of your project and look for `Uno.WinUI`, e.g.

Reviewer: ```suggestion
To see which version of Uno you're currently using, open the `Directory.Packages.props` file at the root of your project and look for `Uno.WinUI`, e.g.
```

```xml
<PackageVersion Include="Uno.WinUI" Version="5.0.116" />
```

> If you have version 5.0.116 (or later) then there's no need to update your version of .NET or Uno.

Reviewer: ```suggestion
If you have version 5.0.116 (or later), then there's no need to update your version of .NET or Uno.
```

## Ensure you are using either .NET 8 (or later) or Uno.WinUI 5.0.116 (or later)

> C# DevKit requires Uno Platform extension 0.12 or later, and the projects need .NET 8 (or later) or use Uno Platform 5.0.116 (or later).

Author: .NET 8 is actually needed to **run** the extension (without Uno 5.0.116), IOW the project(s) can be `net7.0` as long as the extension runs on .NET 8
